From: Dominik George <nik@naturalnet.de>
Subject: Use xorgxrdp instead of x11rdp
--- a/sesman/sesman.ini
+++ b/sesman/sesman.ini
@@ -80,7 +80,7 @@ SyslogLevel=DEBUG
 ; Arch Linux            :  param=/usr/lib/xorg-server/Xorg
 ; CentOS 7              :  param=/usr/bin/Xorg or param=Xorg
 ;
-param=Xorg
+param=/usr/lib/xorg/Xorg
 ; Leave the rest paramaters as-is unless you understand what will happen.
 param=-config
 param=xrdp/xorg.conf
