From: Thorsten Glaser <tg@mirbsd.org>
Subject: Enable Large File Support on (at least) 32-bit Linux
Forwarded: https://github.com/neutrinolabs/xrdp/issues/647
Reviewed-by: Dominik George <nik@naturalnet.de>
--- a/configure.ac
+++ b/configure.ac
@@ -6,6 +6,8 @@ AC_CONFIG_HEADERS(config_ac.h:config_ac-
 AM_INIT_AUTOMAKE([1.7.2 foreign])
 AC_CONFIG_MACRO_DIR([m4])
 AC_PROG_CC
+AC_SYS_LARGEFILE
+AC_FUNC_FSEEKO
 AC_C_CONST
 AC_PROG_LIBTOOL
 
@@ -370,6 +372,7 @@ AC_CONFIG_FILES([
   xup/Makefile
 ])
 
+CPPFLAGS="-include config_ac.h $CPPFLAGS"
 AC_OUTPUT
 
 echo ""
--- a/libpainter/configure.ac
+++ b/libpainter/configure.ac
@@ -6,6 +6,8 @@ AC_CONFIG_HEADERS(config_ac.h:config_ac-
 AM_INIT_AUTOMAKE([1.6 foreign])
 AC_CONFIG_MACRO_DIR([m4])
 AC_PROG_CC
+AC_SYS_LARGEFILE
+AC_FUNC_FSEEKO
 AC_C_CONST
 AC_PROG_LIBTOOL
 PKG_INSTALLDIR
@@ -31,4 +33,5 @@ AC_CONFIG_FILES([
   tests/Makefile
 ])
 
+CPPFLAGS="-include config_ac.h $CPPFLAGS"
 AC_OUTPUT
--- a/librfxcodec/configure.ac
+++ b/librfxcodec/configure.ac
@@ -7,6 +7,8 @@ AM_INIT_AUTOMAKE([1.6 foreign])
 AC_CONFIG_MACRO_DIR([m4])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 AC_PROG_CC
+AC_SYS_LARGEFILE
+AC_FUNC_FSEEKO
 AC_C_CONST
 AC_PROG_LIBTOOL
 PKG_INSTALLDIR
@@ -53,5 +55,6 @@ AC_CONFIG_FILES([Makefile
                  rfxcodec-uninstalled.pc
 ])
 
+CPPFLAGS="-include config_ac.h $CPPFLAGS"
 AC_OUTPUT
 
