From: Thorsten Glaser <tg@mirbsd.org>
Subject: Build system fixes
Forwarded: https://github.com/neutrinolabs/xrdp/pull/467
--- a/xrdp/Makefile.am
+++ b/xrdp/Makefile.am
@@ -23,7 +23,7 @@ endif
 if XRDP_RFXCODEC
 AM_CPPFLAGS += -DXRDP_RFXCODEC
 AM_CPPFLAGS += -I$(top_srcdir)/librfxcodec/include
-XRDP_EXTRA_LIBS += $(top_builddir)/librfxcodec/src/.libs/librfxencode.a
+XRDP_EXTRA_LIBS += $(top_builddir)/librfxcodec/src/.libs/librfxencode.la
 endif
 
 if XRDP_PIXMAN
--- a/xrdpapi/Makefile.am
+++ b/xrdpapi/Makefile.am
@@ -13,3 +13,6 @@ module_LTLIBRARIES = \
 libxrdpapi_la_SOURCES = \
   xrdpapi.c \
   xrdpapi.h
+
+libxrdpapi_la_LIBADD = \
+  $(top_builddir)/common/libcommon.la
