From: Thorsten Glaser <tg@mirbsd.org>
Subject: don’t spit on the console when starting
--- a/common/log.c
+++ b/common/log.c
@@ -382,11 +382,6 @@ internal_config_read_logging(int file, s
     /* try to create path if not exist */
     g_create_path(lc->log_file);
 
-    g_printf("logging configuration:\r\n");
-    g_printf("\tLogFile:       %s\r\n", lc->log_file);
-    g_printf("\tLogLevel:      %i\r\n", lc->log_level);
-    g_printf("\tEnableSyslog:  %i\r\n", lc->enable_syslog);
-    g_printf("\tSyslogLevel:   %i\r\n", lc->syslog_level);
     return LOG_STARTUP_OK;
 }
 
--- a/xrdp/xrdp.c
+++ b/xrdp/xrdp.c
@@ -597,7 +597,6 @@ main(int argc, char **argv)
                    or systemd cannot detect xrdp daemon couldn't start properly */
                 g_exit(1);
             }
-            g_writeln("daemon process %d started ok", pid);
             /* exit, this is the main process */
             g_deinit();
             g_exit(0);
