Subject: Fixes to systemd service files
From: Dominik George <nik@naturalnet.de>
Forwarded: https://github.com/neutrinolabs/xrdp/pull/646
--- a/instfiles/xrdp-sesman.service.in
+++ b/instfiles/xrdp-sesman.service.in
@@ -7,7 +7,8 @@ BindsTo=xrdp.service
 
 [Service]
 Type=forking
-PIDFile=@localstatedir@/run/xrdp-sesman.pid
+PIDFile=@localstatedir@/run/xrdp/xrdp-sesman.pid
+RuntimeDirectory=xrdp
 EnvironmentFile=-@sysconfdir@/sysconfig/xrdp
 EnvironmentFile=-@sysconfdir@/default/xrdp
 ExecStart=@sbindir@/xrdp-sesman $SESMAN_OPTIONS
--- a/instfiles/xrdp.service.in
+++ b/instfiles/xrdp.service.in
@@ -6,9 +6,14 @@ After=network.target xrdp-sesman.service
 
 [Service]
 Type=forking
-PIDFile=@localstatedir@/run/xrdp.pid
+PIDFile=@localstatedir@/run/xrdp/xrdp.pid
+RuntimeDirectory=xrdp
 EnvironmentFile=-@sysconfdir@/sysconfig/xrdp
 EnvironmentFile=-@sysconfdir@/default/xrdp
+User=xrdp
+Group=xrdp
+PermissionsStartOnly=true
+ExecStartPre=/bin/sh /usr/share/xrdp/socksetup
 ExecStart=@sbindir@/xrdp $XRDP_OPTIONS
 ExecStop=@sbindir@/xrdp $XRDP_OPTIONS --kill
 
